src/types.ts

```diff
@@ -135,3 +135,34 @@ export interface RiveRendererInterface {
     androidRenderer: RiveRendererAndroid
   ): void;
 }
+
+export interface FileAssetSource {
+  sourceUrl?: string;
+  sourceAsset?: string;
+  sourceAssetId?: string;
+  path?: string;
+}
```

Comment:
Is it worth being explicit that exactly one of `sourceUrl`, `sourceAsset` and `path` is required?

```
type FileAssetSource =
  | {
      sourceUrl: string
      sourceAsset?: never
      path?: never
      sourceAssetId: string
    }
  | {
      sourceAsset: string
      sourceUrl?: never
      path?: never
      sourceAssetId: string
    }
  | {
      path: string
      sourceUrl?: never
      sourceAsset?: never
      sourceAssetId: string
    }
    ```

Reply:
You mean as a doc comment? Or do you have a suggestion code wise?

This as TypeScript should ensure that only one of sourceUrl, sourceAsset, or path is provided at a time

Reply:
With the current code, it would still be valid if `sourceUrl`, `sourceAsset`, `path`, and `sourceAssetId` were all undefined. It would also be valid if all 4 were defined. 

If you want the type to expect exactly 1 of the 4 values, you might write it like this: 

```
export type FileAssetSource = 
  | { sourceUrl: string; sourceAsset?: never; sourceAssetId?: never; path?: never }
  | { sourceAsset: string; sourceUrl?: never; sourceAssetId?: never; path?: never }
  | { sourceAssetId: string; sourceUrl?: never; sourceAsset?: never; path?: never }
  | { path: string; sourceUrl?: never; sourceAsset?: never; sourceAssetId?: never };
```

Reply:
Ahh sorry @lancesnider, I thought you were quoting code above that was already there 🤦 . Yes, this makes sense to add

Reply:
@lancesnider I played around with this, and this might have other unintended side effects.

The `source_` props are not intended to be set directly by the user. I should have actually named the `path` above `sourcePath`. They will be mapped from the user supplied value.

So if a user sets any of the `source_` options it will fail to work. For simplicity let's keep it as is, and make this a documentation concern.

Reply:
Yep, this was definitely a nitpick. Let's not worry about it. 👍